src/poetry/installation/operations/operation.py

```diff
@@ -1,10 +1,13 @@
 from typing import TYPE_CHECKING
 from typing import Optional
+from typing import TypeVar
 
 
 if TYPE_CHECKING:
     from poetry.core.packages.package import Package
 
+T = TypeVar("T", bound="Operation")
```

Comment:
For someone less experienced with Python typing -- what purpose does the change to `TypeVar` serve here?

Reply:
Great question! Prior to this change:

```
$ pre-commit run -a mypy
src/poetry/puzzle/transaction.py:58: error: Argument 1 to "append" of "list" has incompatible type "Operation"; expected "Union[Install, Uninstall, Update]"
Found 1 error in 1 file (checked 153 source files)
```

The function is typed as literally returning an instance of `Operation`, but in reality it returns `self`. The implications are that `mypy` gets confused for subclasses of `Operation`. See https://mypy.readthedocs.io/en/stable/kinds_of_types.html#the-type-of-class-objects and much prior https://github.com/python/mypy/issues/1212

Reply:
I see! The more you know!